From b7f6edef591562ba44be8b2220503e7ef6b15682 Mon Sep 17 00:00:00 2001
From: Andrej Rosano <andrej@inversepath.com>
Date: Mon, 20 Jun 2016 17:39:01 +0200
Subject: [PATCH] Disable CLI

This patch prevents the falling back to CLI if the boot of
the FIT image fails.
From the CLI it is possible to hot patch the verified boot code
to bypass the FIT image verification.

Signed-off-by: Andrej Rosano <andrej@inversepath.com>
---
 board/inversepath/usbarmory/usbarmory.c | 31 +++++++++++++++++++++++++++++++
 configs/usbarmory_defconfig             |  1 +
 include/configs/usbarmory.h             |  6 ++++++
 3 files changed, 38 insertions(+)

diff --git a/board/inversepath/usbarmory/usbarmory.c b/board/inversepath/usbarmory/usbarmory.c
index a809039..c875e78 100644
--- a/board/inversepath/usbarmory/usbarmory.c
+++ b/board/inversepath/usbarmory/usbarmory.c
@@ -415,3 +415,34 @@ int checkboard(void)
 	puts("Board: Inverse Path USB armory MkI\n");
 	return 0;
 }
+
+#ifndef CONFIG_CMDLINE
+static char *ext2_argv[] = {
+	"ext2load",
+	"mmc",
+	"0:1",
+	USBARMORY_FIT_ADDR,
+	USBARMORY_FIT_PATH
+};
+
+static char *bootm_argv[] = {
+	"bootm",
+	USBARMORY_FIT_ADDR
+};
+
+int board_run_command(const char *cmdline)
+{
+	printf("%s %s %s %s %s\n", ext2_argv[0], ext2_argv[1], ext2_argv[2],
+	       ext2_argv[3], ext2_argv[4]);
+
+	if (do_ext2load(NULL, 0, 5, ext2_argv) != 0) {
+		udelay(5*1000*1000);
+		return 1;
+	}
+
+	printf("%s %s\n", bootm_argv[0], bootm_argv[1]);
+	do_bootm(NULL, 0, 2, bootm_argv);
+
+	return 1;
+}
+#endif
diff --git a/configs/usbarmory_defconfig b/configs/usbarmory_defconfig
index 9f530d9..a52bd3d 100644
--- a/configs/usbarmory_defconfig
+++ b/configs/usbarmory_defconfig
@@ -26,3 +26,4 @@ CONFIG_FIT_SIGNATURE=y
 CONFIG_DM=y
 CONFIG_RSA=y
 CONFIG_RSA_SOFTWARE_EXP=y
+CONFIG_CMDLINE=n
diff --git a/include/configs/usbarmory.h b/include/configs/usbarmory.h
index e3ee39e..f22bbe7 100644
--- a/include/configs/usbarmory.h
+++ b/include/configs/usbarmory.h
@@ -101,6 +101,12 @@
 	"console=ttymxc0,115200\0"				\
 	BOOTENV
 
+#ifndef CONFIG_CMDLINE
+#define CONFIG_BOOTARGS "console=ttymxc0,115200 root=/dev/mmcblk0p1 rootwait rw"
+#define USBARMORY_FIT_PATH     "/boot/usbarmory.itb"
+#define USBARMORY_FIT_ADDR     "0x70800000"
+#endif
+
 /* Physical Memory Map */
 #define CONFIG_NR_DRAM_BANKS		1
 #define PHYS_SDRAM			CSD0_BASE_ADDR
-- 
2.9.0

